packages/network-controller/src/rpc-service/rpc-service.ts

```diff
@@ -483,20 +483,48 @@ export class RpcService implements AbstractRpcService {
     return await this.#policy.execute(async () => {
       const response = await this.#fetch(this.endpointUrl, fetchOptions);
 
-      if (response.status === 405) {
+      if (response.status === 401) {
+        // code: -33100
+        const errorData: JsonRpcError = {
+          code: -33100,
+          message: 'Unauthorized.',
+        };
+        const error = new Error('Unauthorized.');
+        Object.assign(error, errorData);
+        throw error;
+      }
+
+      if (response.status === 402 || response.status === 404) {
+        // code: -32002
+        throw rpcErrors.resourceUnavailable();
```

Comment:
@Gudahtt using `-32002` here will mean it will be considered as a service failure

Reply:
Hmm. Not ideal. We should try to exclude this from consideration by the circuit breaker if we can.

Maybe we can do something like this:

```suggestion
        throw rpcErrors.resourceUnavailable({ message: response.statusText, data: { code: response.status } });
```

Then check `data.code` in the circuit breaker handler to exclude these

Reply:
Or using the conventions you used in the other case, `httpStatus`, `httpStatusText`, and `originalError` for the data properties. Sorry I hadn't read that far when I made this comment

Reply:
Alternatively we could remove this case, and let it get caught by the 4XX fallback block. We can add this later